scripts/src/main/resources/scripts/command/java

```diff
@@ -96,10 +92,197 @@ function doCreate() {
       cd "${DEVON_IDE_HOME}/workspaces/main" || exit 255
     fi
   fi
-  doMavenArchetype "${@}"
+
+  local framework
+  local mavenargs
+  local package
+  local artifactId
+  while [[ -n "${1}" ]]
+  do
+    if [[ "${1}" == "-f" ]] || [[ "${1}" == "-framework" ]]  
+    then
+      framework=${2}
+      shift
+    elif [[ "${1:0:8}" == "-Ddbtype" ]]
+    then 
+      dbtype=${1:9}
+    elif [[ "${1:0:2}" == "-D" ]]
+    then
+      mavenargs="${mavenargs} ${1}"
+    elif [[ "${1}" =~ [a-z]+[.][a-z]+([.][a-z]+)* ]]
+    then 
+      package="${package} ${1}"
+    elif [[ "${1}" =~ [^.]* ]]
+    then
+      ## some better definition for a artifactID TODO
+      artifactId+=${1}
+    fi
+    shift
+  done
+
+  while [[ -z "${framework}" ]] || [[ "${framework}" != "spring" ]] && [[ "${framework}" != "quarkus" ]]
+  do
+    if [[ -n ${package} ]]
+    then
+      framework="spring"
+      break
+    fi
+    if [[ -z "${framework}" ]]
+    then
+      doEcho "No framework selected." 
+    fi
+    read -r -p "(Select a framework: 'quarkus' or 'spring'): " framework
+  done
+
+  if [[ "${framework}" = "quarkus" ]]
+  then
+    startQuarkusProject "${dbtype}"
+  elif [[ "${framework}" = "spring" ]]
+  then
+    if [[ -z "${package}" ]] && [[ -z "${mavenargs}"  ]]
+    then 
+      doEcho "No package or other parameter passed, we need at least a package to create a spring application more information https://devonfw.com/website/pages/docs/devonfw-guide_ide.wiki_cli.asciidoc.html#devonfw-guide_ide.wiki_java.asciidoc_create" 
+      read -r -p "Pass a package and optional parameters as shown in the examples: " mavenargs
+    fi
+    doMavenArchetype "${package}" "${mavenargs}" "${dbtype}" "${artifactId}"
+  fi
   exit ${?}
 }
 
+# dbtype
+function startQuarkusProject() {
+  local message='\nPlease select your preferred extensions and set the project group-, artifactId-, and version
+  on the opened site https://code.quarkus.io/ or just press return to use a configured project from our 
+  templates guides with some extensions that we recommend to start with \n' 
+  while [[ ! "$url" =~ ^https://code.quarkus.io/.* ]]
+  do
+    doEchoAttention -e "$message"
+    read -r -p "(Copy the url and paste it here) The url has to start like this:https://code.quarkus.io/... or just press return to use our simple REST API template project" url
+    if [[  -z "${url}" ]]
+    then 
+      url="https://code.quarkus.io/?g=com.devonfw.quarkus&e=resteasy&e=resteasy-jackson&e=hibernate-validator&e=hibernate-orm"
+      break
+    fi
+  done
+
+  local artifact
+  artifact="$(echo "${url}" | grep -o 'a=[^\&]*')"
+  if [[ -z "${artifact}" ]]
+  then 
+    artifact="myapp"
+    url="${url}&a=${artifact}"
+  else
+    artifact=${artifact:2}
```

Comment:
~this look incorrect to me. The grep will not do any replacement. So if `a=` was found the `artifact` variables is the entire URL and here we only strip the first 2 characters (most likely `ht`) off.
IMHO line 169 needs to be changed to select the artifact from the URL. Might even be doable with pure bash replacements without external processes like grep and sed, but also no problem if we use such processes/tools.~

OK, `grep -o` will do what I expected so already correct...